### ISSUE TITLE
format of OSC messages to radio.js

### ISSUE THREAD

**Issue by thesoundofthesound** [M1Z1/Radio-Crawler#1]
Hi, this code is super interesting and I'm really interested in trying to get it to work with PureData, by sending OSC messages from there. 

I'm just wondering what the formatting of a typical OSC message is that radio.js expects. I'm not familiar with SuperCollider so I'm finding it a little difficult to figure out what's going on there. Any help/insights would be greatly appreciated!

Thanks

**Comment by M1Z1** [comment]
Hello! I'm glad you're interested in the project, it should be fairly easy to run it in PD, I briefly tested it in Max:

Run the script and make sure browser is open and the radios are playing.

The OSC address is 127.0.0.1, 57121

create a udpsend (or equivalent) with the above address.
The message you send should have the following syntax:
/fromPD voice $1 volume $1

You must use the header (/fromPD) and specify the voice, after that you can send any parameter you want and you can even send multiple parameters in a single message

The available parameters are:
volume $1 (-inf, 0 db)
mode $1  (CW, LSB, USB, FM, AM, AMsync)
frequency $1 (float)
mute $1 (0-1)
limiter $1 (-inf, 0db)
solo $1 (0-1)

a message example:

/fromPD voice 1 volume 0 frequency 600 

it will set the first voice (2nd tab) to 0 db volume and frequency 600

From here on the world is your oyster, you can create sequencers, envelopes and whatever else you want inside PD and control Crawler your way! :D 

Let me know if you get it to work, I'm delighted that somebody showed interest, let's talk more!

**Comment by thesoundofthesound** [comment]
This is brilliant. Thank you so much for this info. I got the script running just fine but the thing thing that was stumping me was the 'voice $1' at the start of the OSC message. I've got it working now in a very basic way.  

I'd like to also implement a way of opening and closing the browser with OSC messages from PD. I can probably figure that out from reading up on how to use Puppeteer. 

This code is absolutely great. Thanks so much for working on this, it's a great thing to be able to play with!
